### DISCUSSION TITLE
apache rewrite url based on user id token claim information

### DISCUSSION THREAD

**Discussion by Kurt57** [OpenIDC/mod_auth_openidc#861] [Q&A]
Hi,

I try to figure out how I can make a redirection based on an claim information.

I have three types of this claim property. 

Is it possible for example to write an IF or RewriteCond with "claim"?

I trieft with different location sections with different Require claim:A than same with B than with C etc. but than realized that different sections for same location is not working.

what I have done currently:

```
<Location /foo/bar/oidc>
AuthType openid-connect

Require claim SomeCustomProperty:A
Require claim SomeCustomProperty:B
Require claim SomeCustomProperty:C

RewriteCond %{QUERY_STRING} !SomeCustomProperty=A
 -> SOMETHING LIKE THAT I NEED RewriteCond %{claim} SomeCustomProperty:A
RewriteRule ^/?(.*) https://e2e2-apps.whatsappsim.de/apps2mce/services/oidc/account/netinfo?SomeCustomProperty =A [NE,L]

RewriteCond %{QUERY_STRING} !SomeCustomProperty=B
 -> SOMETHING LIKE THAT I NEED RewriteCond %{claim} SomeCustomProperty:B
RewriteRule ^/?(.*) https://e2e2-apps.whatsappsim.de/apps2mce/services/oidc/account/netinfo?SomeCustomProperty =B [NE,L]

RewriteCond %{QUERY_STRING} !SomeCustomProperty=C
 -> SOMETHING LIKE THAT I NEED RewriteCond %{claim} SomeCustomProperty:B
RewriteRule ^/?(.*) https://e2e2-apps.whatsappsim.de/apps2mce/services/oidc/account/netinfo?SomeCustomProperty =C [NE,L]

</Location>

```

As you see I only want to redirect to same url after login successfully with adding the custom property to URL. So that iOS apps web view can read that to do something special on A B or C

Regards,
Fatih

**Comment by zandbelt** [comment]
yes, you should be able to use `RewriteCond %{ENV:OIDC_CLAIM_SomeCustomProperty} ^A$`

**Reply by gruentee** [reply]
How to handle array values here, for example `realm_access.roles` ?
